src/config.c

```diff
@@ -2853,6 +2854,84 @@ static sds getConfigNotifyKeyspaceEventsOption(standardConfig *config) {
     return keyspaceEventsFlagsToString(server.notify_keyspace_events);
 }
 
+static int setConfigTrustedAddresses(standardConfig *config, sds *argv, int argc, const char **err) {
+    UNUSED(config);
+    int j;
+    int skip = 0;
+
+    if ((unsigned int)argc > server.maxclients || server.trustedIPCount > server.maxclients) {
+        *err = "Too many addresses specified.";
+        return 0;
+    }
+
+    /* A single empty argument is treated as a zero address count */
+    if (argc == 1 && sdslen(argv[0]) == 0) argc = 0;
+
+    for (j = 0; j < argc; j++) {
+        const char *ip = zstrdup(argv[j]);
+        in_addr_t addr = inet_addr(ip);
+        if (addr == 0 || addr == INADDR_NONE) {
+            if (!ip) sds_free((void *)ip);
+            *err = "Invalid address is specified.";
+            return 0;
+        }
+        if (server.trustedIPCount && checkTrustedIP(addr)) {
+            serverLog(LL_NOTICE, "Do not add, IP is exist in the list");
+            skip++;
+            continue;
+        }
+        server.trustedIPList =
+            zrealloc(server.trustedIPList, sizeof(in_addr_t) * (server.trustedIPCount + j - skip + 1));
+        server.trustedIPList[j + server.trustedIPCount - skip] = addr;
+        sds_free((void *)ip);
+    }
+    server.trustedIPCount = server.trustedIPCount + j - skip;
+    valkeySortIP(server.trustedIPList, server.trustedIPCount);
```

Comment:
I think maybe it is a good idea to apply the sort and search logic only when the list size exceeds some size? I would normally expect very small amount of subnets to be defined (not explicit ip addresses)